handler.go

```diff
@@ -0,0 +1,1926 @@
+		return
+	}
+	if err := s.apply(r.Context(), &joinClusterCmd{
+		Cluster: cluster,
+		Node:    addr,
+	}); err != nil {
+		api.Fail(w, err)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *Server) handleShrinkCluster(w http.ResponseWriter, r *http.Request, v api.Verifier) {
+	if _, err := v.Verify(r); err != nil {
+		api.Fail(w, err)
+		return
+	}
+
+	var req api.ShrinkClusterRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		api.Fail(w, kes.NewError(http.StatusBadRequest, err.Error()))
+		return
+	}
+	addr, err := ParseAddr(req.NodeAddr)
+	if err != nil {
+		api.Fail(w, err)
+		return
+	}
+
+	s.mu.RLock()
+	cluster := maps.Clone(s.cluster)
+	s.mu.RUnlock()
+
+	if _, ok := cluster.Lookup(addr); !ok {
+		api.Fail(w, kes.NewError(http.StatusConflict, "node is not part of the cluster"))
+		return
+	}
+	if err = s.apply(r.Context(), &leaveClusterCmd{
+		Cluster: cluster,
+		Node:    addr,
+	}); err != nil {
+		api.Fail(w, err)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *Server) handleDescribeCluster(w http.ResponseWriter, r *http.Request, v api.Verifier) {
+	if _, err := v.Verify(r); err != nil {
+		api.Fail(w, err)
+		return
+	}
+
+	s.mu.RLock()
+	leader, cluster := s.leaderID, maps.Clone(s.cluster)
+	s.mu.RUnlock()
+
+	nodes := make(map[uint64]string, len(cluster))
+	for id, addr := range cluster {
+		nodes[uint64(id)] = addr.String()
+	}
+
+	w.Header().Set(header.ContentType, header.ContentTypeJSON)
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(api.DescribeClusterResponse{
+		Nodes:  nodes,
+		Leader: uint64(leader),
+	})
+}
+
+func (s *Server) handleRestore(w http.ResponseWriter, r *http.Request, v api.Verifier) {
+	if _, err := v.Verify(r); err != nil {
+		api.Fail(w, err)
+		return
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	tmp, err := os.OpenFile(filepath.Join(s.path, "kes.db.new"), os.O_CREATE|os.O_TRUNC|os.O_WRONLY|os.O_SYNC, 0o644)
+	if err != nil {
+		api.Fail(w, err)
+		return
+	}
+	defer tmp.Close()
+
+	if _, err = io.Copy(tmp, r.Body); err != nil {
+		api.Fail(w, err)
+		return
+	}
+	if err = tmp.Sync(); err != nil {
+		api.Fail(w, err)
+		return
+	}
+	if err = tmp.Close(); err != nil {
+		api.Fail(w, err)
+		return
+	}
+
+	if err = s.db.Close(); err != nil {
+		api.Fail(w, err)
+		return
+	}
+	if err := os.Rename(filepath.Join(s.path, "kes.db.new"), filepath.Join(s.path, "kes.db")); err != nil {
+		api.Fail(w, err)
+		return
+	}
+
+	s.db, err = bolt.Open(filepath.Join(s.path, "kes.db"), 0o644, &bolt.Options{
+		FreelistType: bolt.FreelistMapType,
+		Timeout:      3 * time.Second,
+	})
+	if err != nil {
+		api.Fail(w, err)
+		return
+	}
+
+	rootKey, commit, err := initState(s.ctx, s.db, s.hsm)
+	if err != nil {
+		api.Fail(w, err)
+		return
+	}
+	s.rootKey, s.commit = rootKey, commit
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *Server) handleSnapshot(w http.ResponseWriter, r *http.Request, v api.Verifier) {
```

Comment:
Is it like whole bold DB content written as response data? Please add some comments to explain.